tools/build/cross-build/include/common/sys/cdefs.h

```diff
@@ -116,15 +116,33 @@
 #define	__predict_false(exp)    __builtin_expect((exp), 0)
 #endif
 
-#ifndef __weak_reference
 #ifdef __ELF__
+#ifndef __weak_reference
 #define	__weak_reference(sym,alias)	\
 	__asm__(".weak " #alias);	\
 	__asm__(".equ "  #alias ", " #sym)
+#endif
+#ifndef __sym_compat
+#define	__sym_compat(sym,impl,verid)	\
+	__asm__(".symver " #impl ", " #sym "@" #verid)
+#endif
+#ifndef __sym_default
+#define	__sym_default(sym,impl,verid)	\
+	__asm__(".symver " #impl ", " #sym "@@@" #verid)
+#endif
+#ifndef __warn_references
+#define	__warn_references(sym,msg)	\
+	__asm__(".section .gnu.warning." #sym);	\
+	__asm__(".asciz \"" msg "\"");	\
+	__asm__(".previous")
+#endif
```

Comment:
Do we actually need valid definitions of these?

Reply:
Probably not, I guess I could define them as empty for both ELF and MACHO